File: purchasely/android/src/main/kotlin/io/purchasely/purchasely_flutter/NativeViewFactory.kt
package io.purchasely.purchasely_flutter

import android.content.Context
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class NativeViewFactory : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        @Suppress("UNCHECKED_CAST")
        val creationParams = args as Map<String?, Any?>?
        // The inline view surfaces its outcome through the plugin's shared
        // `purchasely-presentation-events` sink (see NativeView), not a dedicated
        // MethodChannel, so no per-view channel is needed.
        return NativeView(context, viewId, creationParams)
    }

    companion object {

        const val VIEW_TYPE_ID = "io.purchasely.purchasely_flutter/native_view"
    }
}

File: purchasely/ios/purchasely_flutter/Classes/NativeViewFactory.swift
import Foundation
import Flutter
import UIKit
import Purchasely

class NativeViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger

    init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
        super.init()
    }

    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        // The inline view surfaces its outcome through the plugin's shared
        // `purchasely-presentation-events` sink (see NativeView), not a dedicated
        // MethodChannel, so no per-view channel is needed.
        return NativeView(
            frame: frame,
            viewIdentifier: viewId,
            arguments: args)
    }

    /// Implementing this method is only necessary when the `arguments` in `createWithFrame` is not `nil`.
    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
          return FlutterStandardMessageCodec.sharedInstance()
    }
}
